parlai/core/params.py

```diff
@@ -106,12 +109,17 @@ def add_model_args(self):
             '-m', '--model', default='repeat_label',
             help='the model class name, should match parlai/agents/<model>')
         self.parser.add_argument(
-            '-mp', '--model_params', default='',
-            help='the model parameters, a string that is parsed separately '
-            + 'by the model parser after the model class is instantiated')
-        self.parser.add_argument(
             '-mf', '--model_file', default='',
             help='model file name for loading and saving models')
+        # Find which model specified, and add its specific arguments.
+        model = None
+        for index, item in enumerate(sys.argv):
+            if item == '-m' or item == '--model':
+                model = sys.argv[index + 1]
+        if model:
+            agent = get_agent_module(model)
+            if hasattr(agent, 'add_cmdline_args'):
+                agent.add_cmdline_args(self)
```

Comment:
could add `add_cmdline_args` to core/agents.py:Agent so that there's an empty definition instead of checking if the definition exists, but this is fine too